sentry/src/main/java/io/sentry/ServerSessionManager.java

```diff
@@ -0,0 +1,52 @@
+package io.sentry;
+
+import io.sentry.util.Objects;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class ServerSessionManager implements SessionTracker, SessionUpdater {
+  private @Nullable Status status;
+  private final @NotNull SessionAggregates sessionAggregates;
+
+  ServerSessionManager(final @NotNull SessionAggregates sessionAggregates) {
+    this.sessionAggregates =
+        Objects.requireNonNull(sessionAggregates, "sessionAggregates is required");
+  }
+
+  @Override
+  public void startSession() {
+    status = null;
+  }
+
+  @Override
+  public void endSession() {
+    if (status == null) {
+      status = Status.Exited;
+    }
+    sessionAggregates.addSession(status);
+  }
+
+  @Override
+  public @Nullable Session updateSessionData(
+      @NotNull SentryEvent event, @Nullable Object hint, @Nullable Scope scope) {
+    // If the status is still just 'null' or 'Exited'
+    // perhaps promote it to errored or crashed:
+    if (status != Status.Crashed) {
+      if (event.isCrashed()) {
+        status = Status.Crashed;
+      } else if (event.isErrored()) {
+        status = Status.Errored;
+      }
+    }
+    return null;
+  }
+
+  @ApiStatus.Internal
+  public enum Status {
```

Comment:
These statuses are slightly different, so AFAIR i did not want to couple these parts. Since its internal, it can be a subject of further refactoring if needed.